kyo-data/shared/src/main/scala/kyo/Result.scala

```diff
@@ -32,10 +58,56 @@ object Result:
 
     inline def catching[E <: Throwable]: CatchingOps[E] = CatchingOps(())
 
-    inline def success[E, A](inline value: A): Result[E, A]           = Success(value)
-    inline def fail[E, A](inline error: E): Result[E, A]              = Fail(error)
+    /** Creates a successful Result.
+      *
+      * @param value
+      *   The successful value
+      * @tparam E
+      *   The type of the error (not used in this case)
+      * @tparam A
+      *   The type of the successful value
+      * @return
+      *   A successful Result
+      */
+    inline def success[E, A](inline value: A): Result[E, A] = Success(value)
```

Comment:
style: I think this should be renamed to `succeed` (perhaps I am too used to ZIO)

Reply:
let's follow up separately